server/lib/genome/groupers/base.rb

```diff
@@ -30,10 +53,37 @@ def fetch_source_meta
         body['source_matches'].transform_values { |value| value['source_meta_'] }
       end
 
-      # Normalize claim terms
-      # If the primary term normalizes successfully, go with that. Otherwise, iterate
-      # through claims and tally all 'votes' for normalized concepts with the highest match
-      # type. For example:
+      # Construct individual groups for normalizing terms
+      # Returns a sorted array of normalizable concept IDs, a primary name if given,
+      # and aliases that aren't otherwise known to normalize (e.g. other names, trade names,
+      # concept IDs from niche namespaces)
+      def get_claim_search_groups(claim, claim_aliases)
+        groups = { 'concept_ids' => [], 'name' => nil, 'aliases' => [] }
+
+        concept_ids = []
+        if normalizable_ids.include? claim.nomenclature
+          concept_ids << [claim.name, claim.nomenclature]
+        else
+          groups['name'] = claim.name
+        end
+
+        claim_aliases.each do |ca|
+          if normalizable_ids.include? ca.nomenclature
+            concept_ids << [ca.alias, ca.nomenclature]
+          else
+            groups['aliases'] << ca.alias
+          end
+        end
+
+        groups['concept_ids'] = concept_ids
+                                .sort_by! { |_, nomenclature| normalizable_ids.index(nomenclature) || Float::INFINITY }
```

Comment:
The `!` here will mutate the original array rather than returning a copy. It's entirely possible you intended this, it just reads as a little strange chained with the `map` which will produce a copy so I wanted to flag it in case.